### PR TITLE
network: add hackage remote-id

### PR DESCRIPTION
**Pull request by thesamesam**
Signed-off-by: Sam James <sam@gentoo.org>

### CODE REVIEW

**Comment by thesamesam** [comment]
I can't test this properly, just like #415, as I have an issue installing snakeoil in a venv:
```
Processing /home/sam/git/pkgcheck
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [17 lines of output]
      Traceback (most recent call last):
        File "/home/sam/tmp/venv/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 363, in <module>
          main()
        File "/home/sam/tmp/venv/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 345, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/home/sam/tmp/venv/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 130, in get_requires_for_build_wheel
          return hook(config_settings)
        File "/tmp/pip-build-env-v_99pr84/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 177, in get_requires_for_build_wheel
          return self._get_build_requires(
        File "/tmp/pip-build-env-v_99pr84/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 159, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-v_99pr84/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 174, in run_setup
          exec(code, locals())
        File "<string>", line 15, in <module>
        File "/tmp/pip-build-env-v_99pr84/overlay/lib/python3.10/site-packages/snakeoil/dist/distutils_extensions.py", line 54, in <module>
          raise ImportError('this module is only meant to be imported in setup.py scripts')
      ImportError: this module is only meant to be imported in setup.py scripts
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

**Comment by laumann** [comment]
I can confirm this with the steps to reproduce:

```bash
python -m venv ~/tmp/venv
. ~/tmp/venv/bin/activate
pip install -r requirements/dev.txt # in pkgcheck folder
```
I tried switching to older versions of snakeoil and got _different_ errors, for example v0.9.9:

```
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [19 lines of output]
      Traceback (most recent call last):
        File "/home/tj/sources/gentoo/trash/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 363, in <module>
          main()
        File "/home/tj/sources/gentoo/trash/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 345, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/home/tj/sources/gentoo/trash/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 130, in get_requires_for_build_wheel
          return hook(config_settings)
        File "/tmp/pip-build-env-buomd570/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 177, in get_requires_for_build_wheel
          return self._get_build_requires(
        File "/tmp/pip-build-env-buomd570/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 159, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-buomd570/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 174, in run_setup
          exec(code, locals())
        File "<string>", line 10, in <module>
        File "/home/tj/sources/gentoo/snakeoil/src/snakeoil/dist/distutils_extensions.py", line 34, in <module>
          from setuptools.command import install as dst_install
        File "/tmp/pip-build-env-buomd570/overlay/lib/python3.10/site-packages/setuptools/command/install.py", line 6, in <module>
          import distutils.command.install as orig
      ImportError: cannot import name 'install' from 'setuptools._distutils.command' (/tmp/pip-build-env-buomd570/overlay/lib/python3.10/site-packages/setuptools/_distutils/command/__init__.py)
      [end of output]
```

File: src/pkgcheck/checks/network.py
Comment (hololeap) [inline src/pkgcheck/checks/network.py:277]: This really should be `'hackage': ...`

hackport is just a project of gentoo-haskell, a portmanteau of "hackage" and "portage".

Comment (thesamesam) [reply src/pkgcheck/checks/network.py:277]: I just made a typo.

Comment (arthurzam) [reply src/pkgcheck/checks/network.py:277]: ping?

Comment (thesamesam) [reply src/pkgcheck/checks/network.py:277]: Will fix shortly

Comment (thesamesam) [reply src/pkgcheck/checks/network.py]: done

**Review by arthurzam — APPROVED** [review]
Tested locally, it passes
Just in case, I will wait until new release of `snakeoil` is done before merging, so we can see the master as green

**Pull request closed without merging**
